### DISCUSSION TITLE
Deno has already introduced sqlite in the source code, why not open the corresponding interface

### DISCUSSION THREAD

**Discussion by rzl** [denoland/deno#23502] [Q&A]
Deno has already introduced sqlite in the source code, why not open the corresponding interface

**Comment by jupegarnica** [comment]
🍿

**Comment by jtoppine** [comment]
Not part of the team, just how I interpret earlier discussions about the topic:

Apparently the sqlite in Deno is considered an implementation detail, which could theoretically be changed or upgraded to something else, potentially not backwards compatible in the future. Exposing sqlite in the public API would mean that exact API would need to be kept and maintained indefinitely. 

Additionally; in general Deno attempts to stick to standardized web API's, and keep non-standard runtime features (like embedded sqlite would be) to a minimum.  Not sure but I think Deno supports some web standard db/storage features like localStorage/indexedDB and such (not super popular).

Additionally, and somewhat contrarily to the previous point, Deno already has another, custom embedded db, KV. Which works better with the proprietary Deno Deploy edge runtime service, where it is implemented using FoundationDB. On self-hosted deno, KV runs on top of the embedded sqlite.

I guess the thought process is to limit the amount of built-in storage solution API's to a minimum, and refer to KV or third party modules for other solutions.

All of that being said, although some of the rationale for not exposing it makes sense, I personally would like to see it happen nevertheless. I use sqlite with deno a lot and none of the available third party modules for it are perfect. sqlite via FFI requires wide open permissions, and the WASM version, while having served me well, is not exactly optimal. Being able to just utilize the embedded sqlite would be nice.
